provisioning/state_service.go

```diff
@@ -0,0 +1,58 @@
+package provisioning
+
+import (
+	"fmt"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+)
+
+const (
+	stateService = "metal3-state"
+)
+
+func newMetal3StateService(targetNamespace string) *corev1.Service {
+	port, _ := strconv.Atoi(baremetalHttpPort) // #nosec
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      stateService,
+			Namespace: targetNamespace,
+			Annotations: map[string]string{
```

Comment:
Do we need this annotation for things we create dynamically? I thought this was a signal to the thing that reads the manifests?

Reply:
I have no idea, very much cargo-culting it. @stbenjam?

Reply:
I think CVO is the only one reading this annotation for the resources that it is creating using the files in /manifests. Although, I do see this as the only place it is being added dynamically https://github.com/openshift/machine-api-operator/blob/master/pkg/operator/baremetal_pod.go#L89.

Probably harmless here but not necessary.